Solutions/LastPass/Data Connectors/LastPassAPIConnector.json

```diff
@@ -1,123 +1,132 @@
+            {
+              "name": "{{graphQueriesTableName}}",
+              "lastDataReceivedQuery": "{{graphQueriesTableName}}\n            | summarize Time = max(TimeGenerated)\n            | where isnotempty(Time)"
             }
+          ],
+          "connectivityCriterias": [
+            {
+              "type": "SentinelKindsV2",
+              "value": [
+                "APIPolling"
+              ]
             }
-    ]
-}
\ No newline at end of file
+          ],
+          "availability": {
+            "status": 1,
+            "isPreview": true
+          },
+          "permissions": {
+            "resourceProvider": [
+              {
+                "provider": "Microsoft.OperationalInsights/workspaces",
+                "permissionsDisplayText": "read and write permissions are required.",
+                "providerDisplayName": "Workspace",
+                "scope": "Workspace",
+                "requiredPermissions": {
+                  "action": true,
+                  "write": true,
+                  "read": true,
+                  "delete": true
+                }
+              }
+            ],
+            "customs": [
+              {
+                "name": "LastPass API Key and CID",
+                "description": "A LastPass API key and CID are required. [See the documentation to learn more about LastPass API](https://support.logmeininc.com/lastpass/help/use-the-lastpass-provisioning-api-lp010068)."
+              }
+            ]
+          },
+          "instructionSteps": [
+            {
+              "title": "Connect LastPass Enterprise to Azure Sentinel",
+              "description": "Provide the LastPass Provisioning API Key.",
+              "instructions": [
+                {
+                  "parameters": {
+                    "enable": "true",
+                    "userRequestPlaceHoldersInput": [
+                      {
+                        "displayText": "cid",
+                        "requestObjectKey": "queryParametersTemplate",
+                        "placeHolderName": "{{cidPlaceHolder}}"
+                      }
+                    ]
+                  },
+                  "type": "APIKey"
+                }
+              ]
+            }
+          ]
+        },
+        "pollingConfig": {
+          "auth": {
+            "authType": "APIKey",
+            "APIKeyName": "provhash",
+            "IsAPIKeyInPostPayload": true
+          },
+          "request": {
+            "apiEndpoint": "https://lastpass.com/enterpriseapi.php",
+            "rateLimitQPS": 2,
+            "httpMethod": "Post",
+            "queryTimeFormat": "yyyy-MM-ddTHH:mm:ssZ",
+            "retryCount": 3,
+            "queryWindowInMin": 10,
+            "timeoutInSeconds": 120,
+            "queryParametersTemplate": "{'cid': '{{cidPlaceHolder}}', 'cmd': 'reporting', 'data': { 'from': '{_QueryWindowStartTime}', 'to': '{_QueryWindowEndTime}' }, '{_APIKeyName}': '{_APIKey}'}",
```

Comment:
@ThijsLecomte - I hope you have tested the paging and api functionality to retrieve the data to LA workspace.

Reply:
Yes, I had this running in production for a while.